### PR TITLE
Add app targets and workflow

### PR DESCRIPTION
**Pull request by mcharriere**
- Add update-chart with vendir to Makefile for apps
- Add optional workflow with vendir for app repos 

Towards https://github.com/giantswarm/giantswarm/issues/23766

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: pkg/gen/input/workflows/internal/file/update_chart.yaml.template [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    runs-on: ubuntu-20.04
+    needs:
+      - gather_facts
+    if: ${{ needs.gather_facts.outputs.skip != 'true' }}
+    env:
+      architect_flags: "--organisation ${{ github.repository_owner }} --project ${{ needs.gather_facts.outputs.repo_name }}"
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '=1.18.1'
```
Comment (ubergesundheit) [inline pkg/gen/input/workflows/internal/file/update_chart.yaml.template:75]: do we need go?

File: pkg/gen/input/workflows/internal/file/update_chart.yaml.template [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.gather_facts.outputs.branch }}
+      - name: Install vendir
+        run: |
+          wget https://github.com/vmware-tanzu/carvel-vendir/releases/download/v0.32.0/vendir-linux-amd64 -O /tmp/vendir
+          SHA_VENDIR=`shasum -a 256 /tmp/vendir | cut -d ' ' -f1`
+          [ "$SHA_VENDIR" != "0b52c170f4a30c2b6213ff0048ecc89c9c25c3e4da56eb1e095fcdb335bd82ed" ] && echo "invalid vendir binary" && exit -1
```
Comment (ubergesundheit) [inline pkg/gen/input/workflows/internal/file/update_chart.yaml.template:84]: I would vote to encode version and shasum into variables inside the go code

File: pkg/gen/input/workflows/internal/file/update_chart.yaml.template (hunk 1 above)
Comment (mcharriere) [reply pkg/gen/input/workflows/internal/file/update_chart.yaml.template]: no, it's gone now

File: pkg/gen/input/workflows/internal/file/update_chart.yaml.template (hunk 2 above)
Comment (mcharriere) [reply pkg/gen/input/workflows/internal/file/update_chart.yaml.template]: make sense. done

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Add
```
Comment (ubergesundheit) [inline CHANGELOG.md:10]:
```suggestion
### Added
```

File: pkg/gen/input/workflows/internal/file/update_chart.yaml.template [hunk 4]
```diff
@@ -0,0 +1,99 @@
+{{{{ .Header }}}}
+name: Create Update PR
+on:
+  push:
+    branches:
+      - 'main#update'
+      - 'master#update'
```
Comment (ubergesundheit) [inline pkg/gen/input/workflows/internal/file/update_chart.yaml.template:7]: unsure if branch name `update` is too generic. I would suggest `update_chart` as an improvement.

File: pkg/gen/input/workflows/internal/file/update_chart.yaml.template [hunk 5]
```diff
@@ -0,0 +1,99 @@
+{{{{ .Header }}}}
+name: Create Update PR
```
Comment (ubergesundheit) [inline pkg/gen/input/workflows/internal/file/update_chart.yaml.template:2]: Optional!

I think adding a two or three sentence explanationatory comment at the head of the file would help people stumbling on this file in app repositories.

File: pkg/gen/input/workflows/internal/file/update_chart.yaml.template [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+      - name: Print github context JSON
+        run: |
+          cat <<EOF
+          ${{ toJson(github) }}
+          EOF
+  gather_facts:
+    name: Gather facts
+    runs-on: ubuntu-20.04
+    outputs:
+      repo_name: ${{ steps.gather_facts.outputs.repo_name }}
```
[the remark this answers is not in the document]
Comment (mcharriere) [reply pkg/gen/input/workflows/internal/file/update_chart.yaml.template:32]: `is_major` gone. `repo_name` is needed below now

File: pkg/gen/input/workflows/internal/file/update_chart.yaml.template [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+            base="$(echo $head | cut -d '#' -f 1)"
+          else
+            base="${{ github.event.base_ref }}"
+          fi
+
+          base="${base#refs/heads/}" # Strip "refs/heads/" prefix.
+
+          repo_name="$(echo '${{ github.repository }}' | awk -F '/' '{print $2}')"
+          echo "repo_name=\"$repo_name\" base=\"$base\" head=\"$head\""
+          echo "repo_name=${repo_name}" >> $GITHUB_OUTPUT
```
[the remark this answers is not in the document]
Comment (mcharriere) [reply pkg/gen/input/workflows/internal/file/update_chart.yaml.template:54]: I was missing some logic. thanks for spotting this!

File: pkg/gen/input/workflows/internal/file/update_chart.yaml.template (hunk 4 above)
Comment (mcharriere) [reply pkg/gen/input/workflows/internal/file/update_chart.yaml.template]: I went for `update-chart`, following the make file target

File: pkg/gen/input/workflows/internal/file/update_chart.yaml.template [hunk 8]
```diff
@@ -76,11 +80,30 @@ jobs:
...
+          then
+            echo "Could not find chart directory '${chart}', removing app suffix."
+
+            # Remove app suffix.
+            chart="helm/${repository%-app}"
+
+            if [ ! -d "${chart}" ]
+            then
+              # Print error.
+              echo "Could not find chart directory '${chart}', doing nothing."
```
Comment (ubergesundheit) [inline pkg/gen/input/workflows/internal/file/update_chart.yaml.template:102]: maybe even fail with `exit 1`?

**Review by ubergesundheit — APPROVED** [review]

**Pull request merged**